File: python/simple_interest.py
def simpleInterest(amount:int, rate:float, year:int) -> int:
    if amount == 0 or rate == 0 or year == 0 :
        raise ValueError("Invalid amount,interest or year")
    
    interest = (amount * rate * year) / 100

    return interest



def main():
    print("Welcome to my simple interest calculator")
    amount = int(input("Enter the amount: "))
    rate = float(input("Enter the rate: "))
    year = int(input("Enter the amount of year: "))

    interest = simpleInterest(amount, rate, year)

    print(f"The simple interest is: {interest}")

if __name__ == "__main__":
    main()    

File: typescript/src/simpleInterest.ts
const simpleInterest = (principle:number, rate:number, year:number):number => {
    if(principle === 0 || rate === 0 || year === 0){
        return 0
    }

    const interest:number = (principle * year * rate) / 100

    return interest
}


const principle = 1000;
const rate = 3;
const year = 2;

const interest = simpleInterest(principle,rate,year)

console.log(`Simple interest on this payment is: ${interest}`)